controllers/datamovement_controller.go

```diff
@@ -192,62 +218,175 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, handleInvalidError(err)
 	}
 
+	// Expand the context with cancel and store it in the map so the cancel function can be used in
+	// another reconciler loop. Also add NamespacedName so we can retrieve the resource.
+	ctxCancel, cancel := context.WithCancel(ctx)
+	r.contexts.Store(dm.Name, dataMovementCancelContext{
+		ctx:    ctxCancel,
+		cancel: cancel,
+	})
+
+	// Pull configurable values from the ConfigMap
+	configMap := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: configMapNamespace}, configMap); err != nil {
+		log.Info("Config map not found - requeueing")
+		return ctrl.Result{}, err
+	}
+	log.Info("Config map found:", "configMap", configMap, "data", configMap.Data)
+	dmCmd, progressCollectInterval, dcpProgressInterval, numProcesses := parseConfigMapValues(*configMap)
+
+	// Set the command and arguments according to the values specified by the config map
+	cmdStr, cmdArgs := getCmdAndArgs(dmCmd, numProcesses, dcpProgressInterval, hosts, dm)
+	cmd := exec.CommandContext(ctxCancel, cmdStr, cmdArgs...)
+
 	// Record the start of the data movement operation
 	now := metav1.NowMicro()
 	dm.Status.StartTime = &now
 	dm.Status.State = nnfv1alpha1.DataMovementConditionTypeRunning
+	dm.Status.CommandStatus = &nnfv1alpha1.NnfDataMovementCommandStatus{}
```

Comment:
You're still not logging the command string in the dm.Status.CommandStatus for the below Status().Update()

Reply:
I'll fix.